### PR TITLE
Add support for production build and env parameter

### PR DESCRIPTION
**Pull request by vkhoroz**
Signed-off-by: Volodymyr Khoroz <volodymyr.khoroz@foundries.io>

### CODE REVIEW

**Comment by vkhoroz** [comment]
Example usage:
- Recommended is to build a separate lmp-device-register for production devices:
  `cmake -B build -DDEVICE_FACTORY=default -DDEVICE_API=https://osf-ubuntu.local:2443/sign/ -DGIT_COMMIT=01a09d79 -DHARDWARE_ID=intel-corei7-64 -DDOCKER_COMPOSE_APP=1 -DAKLITE_TAGS=1 -DPROMOTED_TARGETS=1 && ( cd build && make )`
  After that registering a new device is easy:
  `lmp-device-register -n prod-dev1 -d /tmp/sota/prod-dev1 --start-daemon=0 --tags beta`

- For testing purposes it can be used without a separate build like this:
  `cmake -B build -DDEVICE_FACTORY=default -DDEVICE_API=https://osf-ubuntu.local:2443/sign/ -DGIT_COMMIT=01a09d79 -DHARDWARE_ID=intel-corei7-64 -DDOCKER_COMPOSE_APP=1 -DAKLITE_TAGS=1 && ( cd build && make )`
  `PROMOTED_TARGETS=1 lmp-device-register -n prod-dev1 -d /tmp/sota/prod-dev1 --start-daemon=0 --tags beta`

**Comment by doanac** [comment]
I think "promoted targets" is going to be confusing. We need a flag to say "this is a production device". So maybe `-DPRODUCTION_DEVICE=1`

**Comment by vkhoroz** [comment]
> 
> 
> I think "promoted targets" is going to be confusing. We need a flag to say "this is a production device". So maybe `-DPRODUCTION_DEVICE=1`

I think that yes, I'm overextending a term "promoted targets".  For device I agree a simple term "production device" makes more sense.  Although I think there are other use cases when customers might want a decide to behave this way, a primary goal of this feature really goes for production devices.

**Comment by doanac** [comment]
>  I think there are other use cases when customers might want a device to behave this way
I'm thinking those cases are where they want their device to be a 'production' device. They just want to test it before its *really* production.

**Review by mike-sul — APPROVED** [review]
the code LGTM, although I am not sure about the log messages "Enabling production business category", it rather says about internal details of our implementation of a production device certificates generation. Maybe we should log something like "Enabling certificate generation for a production device" or "Enabling generation a production certificate".

**Comment by vkhoroz** [comment]
@doanac please, approve.

**Comment by doanac** [comment]
@vkhoroz - i think i left this trying to get @ricardosalveti input about your ifdef logic. It seems a little strange and I wasn't sure if he's going to want both approaches like that.

**Comment by vkhoroz** [comment]
> @vkhoroz - i think i left this trying to get @ricardosalveti input about your ifdef logic. It seems a little strange and I wasn't sure if he's going to want both approaches like that.
@doanac @ricardosalveti If you find this strange, FYI I've used the same pattern as we use for other arguments.

And if you're concerned about security - there is no additional security if we only have one way, because anybody may pull in lmp-device-register and build it with any arguments from source.  A protection goes from a <token> parameter, not build args,

**Comment by ricardosalveti** [comment]
> > @vkhoroz - i think i left this trying to get @ricardosalveti input about your ifdef logic. It seems a little strange and I wasn't sure if he's going to want both approaches like that.
> 
> @doanac @ricardosalveti If you find this strange, FYI I've used the same pattern as we use for other arguments.
> 
> And if you're concerned about security - there is no additional security if we only have one way, because anybody may pull in lmp-device-register and build it with any arguments from source. A protection goes from a `token` parameter, not build args,

I'm fine to the possibility to enable via environment, just not completely sure if that is a path we would expect to use (unless if something that can be done for testing purposes).

File: src/main.cpp [hunk 1]
```diff
@@ -74,6 +75,19 @@ static void _set_factory_option(std::string& factory) {
...
+#ifdef PRODUCTION
+	is_prod = true;
+#else
+	is_prod = false;
+	const char* production_env_var = std::getenv("PRODUCTION");
+	if (production_env_var != nullptr) {
+		cout << "Enabling production business category via the environment variable" << endl;
+		is_prod = true;
+	}
+#endif
```
[the remark this answers is not in the document]
Comment (ricardosalveti) [reply src/main.cpp:88]: It is fine from the most safe scenario, so +1

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**